include/miopen/miopen.h

```diff
@@ -6582,6 +6583,105 @@ MIOPEN_EXPORT miopenStatus_t miopenBackendInitialize(miopenBackendDescriptor_t d
 // CLOSEOUT BackendAPI DOXYGEN GROUP
 #endif // MIOPEN_BETA_API
 
+#ifdef MIOPEN_BETA_API
+
+/*! @ingroup interpolate
+ * @enum miopenInterpolateMode_t
+ * Modes for Interpolate
+ */
+
+typedef enum
+{
+    MIOPEN_INTERPOLATE_MODE_NEAREST   = 0,
+    MIOPEN_INTERPOLATE_MODE_LINEAR    = 1,
+    MIOPEN_INTERPOLATE_MODE_BILINEAR  = 2,
+    MIOPEN_INTERPOLATE_MODE_BICUBIC   = 3,
+    MIOPEN_INTERPOLATE_MODE_TRILINEAR = 4,
+} miopenInterpolateMode_t;
+
+// Interpolate APIs
+/** @addtogroup interpolate
+ *
+ *  @{
+ */
+
+/*! @brief Execute a interpolate forward layer
+ *
+ * @param handle                MIOpen handle (input)
+ * @param inputDesc             Tensor descriptor for input tensor (input)
+ * @param input                 Data tensor input  (input)
+ * @param outputDesc            Tensor descriptor for output tensor (input)
+ * @param output                Data tensor output (output)
+ * @param scaleFactorsDesc      Tensor descriptor for scale factors tensor (input)
+ * @param scale_factors         Data tensor scale factors (input)
+ * @param mode                  Interpolation mode (input)
+ * @param align_corners         Align corners (input)
+ * @return                      miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t
+miopenInterpolateForward(miopenHandle_t handle,
```

Comment:
Because they are all used to calculate (`scale_factor`) or directly determine (`size`) the size, I think it is redundant, and the kernel code use `scale_factor` only. I retained both `size` and `scale_factor` in the driver layer. Is it okay, or do I need to add the `size` parameter to the API? @kyeonghwanryu

Reply:
I think API should support both scale_factor and size. Just add some lines to calculate scale_factor from size in interpolate_api.cpp, and then everything else in your implementation will still be used scale_factor without any problems